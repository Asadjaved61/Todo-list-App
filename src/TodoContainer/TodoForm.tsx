import React, { useState } from 'react';
import TodoFormProps from '../Interfaces/TodoForm.interface';

const TodoForm: React.FC<TodoFormProps> = ({ addTodo }) => {
    const [value, setValue] = useState<string>("");
  
    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
      e.preventDefault();
      if (!value) return;

      addTodo(value);
    };
  
    return (
      <form className="add-todo-form center-element" onSubmit={handleSubmit}>
        <input
          type="text"
          className="input"
          placeholder="Add todo..."
          value={value}
          onChange={e => setValue(e.target.value)}
          required
        />
        <button type="submit" className="btn btn-submit">Add</button>
      </form>
    );
  }

  export default TodoForm;