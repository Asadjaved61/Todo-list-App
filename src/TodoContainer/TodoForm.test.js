import React from 'react'
import {render, fireEvent, cleanup} from '@testing-library/react';

import TodoForm from './TodoForm';

describe('Todo Container', () => {
    afterEach(cleanup);

    it('it should match snapshot', () => {
        const { asFragment } = render(<TodoForm />);

        expect(asFragment()).toMatchSnapshot();
    })

    it('addTodo should add new todo', async () => {
        const { getByText, getByPlaceholderText, get } = render(<TodoForm />);
        getByPlaceholderText('Add todo...').textContent = 'shopping';
        await fireEvent.click(getByText('Add'));

        expect(getByText('shopping')).toBeTruthy();
    })
})