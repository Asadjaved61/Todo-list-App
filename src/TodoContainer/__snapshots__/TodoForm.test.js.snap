// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Container it should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="add-todo-form center-element"
  >
    <input
      class="input add-todo-input"
      name="addTodo"
      placeholder="Add todo..."
      required=""
      type="text"
      value=""
    />
    <button
      class="btn btn-submit"
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
